ggshield/verticals/secret/output/secret_text_output_handler.py

```diff
@@ -105,15 +104,16 @@ def process_result(
         # policy breaks and matches are modified in the functions leak_dictionary_by_ignore_sha and censor_content.
         # Previously process_result was executed only once, so it did not create any issue.
         # In the future we could rework those functions such that they do not change what is in the result.
-        policy_breaks = deepcopy(result.scan.policy_breaks)
+        result.scan.policy_breaks = deepcopy(result.scan.policy_breaks)
```

Comment:
I am not sure this call to deepcopy() is useful. The goal of the deepcopy() call in the main branch was to ensure we do not modify the policy breaks inside the `result` argument, but this version still modifies it.

In a previous iteration of this branch the code made a deepcopy of the whole `result` instance, which sounds like what we want. What was the issue with that?

Reply:
Making a deep copy of result breaks the functional test `tests/functional/secret/test_scan_docker.py::test_scan_docker_uses_cache` with the error: `Error: cannot pickle 'ExFileObject' object`

Reply:
I see. I think this can be fixed with this:

```python
result = Result(file=result.file, scan=deepcopy(result.scan))
```

This way we get a new result.scan instance which we can modify, but we reuse the existing result.file, which cannot be deepcopied.